### PR TITLE
Feature/1209 device token key

### LINKED ISSUE #1209: Rename `ARTDeviceTokenKey` to `ARTAPNSDeviceTokenKey`
We should rename `ARTDeviceTokenKey` to `ARTAPNSDeviceTokenKey`. It's a small improvement in the naming, but it would avoid confusion. We currently have `ARTDeviceIdentityTokenKey` and `ARTDeviceTokenKey`. One is for APNs, the other one (`ARTDeviceIdentityTokenKey`) is for Ably.

For more information about what `deviceIdentityToken` is, read https://docs.ably.io/client-lib-development-guide/features/#RSH6a.

APNs device token is the token used by Ably with APNs to select the device for push notifications.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1350) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by maratal**
Closes #1209

### CODE REVIEW

File: Source/ARTLocalDevice.m [hunk 1]
```diff
@@ -70,7 +70,7 @@ + (ARTLocalDevice *)load:(NSString *)clientId storage:(id<ARTDeviceStorage>)stor
     ARTDeviceIdentityTokenDetails *identityTokenDetails = [ARTDeviceIdentityTokenDetails unarchive:identityTokenDetailsInfo];
     device->_identityTokenDetails = identityTokenDetails;
 
-    [device setDeviceToken:[storage objectForKey:ARTDeviceTokenKey]];
+    [device setDeviceToken:[storage objectForKey:ARTAPNSDeviceTokenKey]];
```
Comment (lukasz-szyszkowski) [inline Source/ARTLocalDevice.m:73]: This method should be called `setAPNSDeviceToken`

File: Source/ARTLocalDevice.m [hunk 2]
```diff
@@ -94,7 +94,7 @@ - (void)setDeviceToken:(NSString *_Nonnull)token {
 }
 
 - (void)setAndPersistDeviceToken:(NSString *)deviceToken {
```
Comment (lukasz-szyszkowski) [inline Source/ARTLocalDevice.m:96]: I know you didn’t change it but it should be called `setAndPersistAPNSDeviceToken`

File: Source/ARTPush.m [hunk 3]
```diff
@@ -179,7 +179,7 @@ + (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceTokenDa
     NSString *deviceToken = [hexString copy];
 
     [rest.logger info:@"ARTPush: device token: %@", deviceToken];
-    NSString *currentDeviceToken = [rest.storage objectForKey:ARTDeviceTokenKey];
+    NSString *currentDeviceToken = [rest.storage objectForKey:ARTAPNSDeviceTokenKey];
```
Comment (lukasz-szyszkowski) [inline Source/ARTPush.m:182]: Rename it to`currentAPNSDeviceToken`

File: Source/ARTLocalDevice.m (hunk 1 above)
Comment (maratal) [reply Source/ARTLocalDevice.m]: done f9c88fd

File: Source/ARTLocalDevice.m (hunk 2 above)
Comment (maratal) [reply Source/ARTLocalDevice.m]: done f9c88fd

File: Source/ARTPush.m (hunk 3 above)
Comment (maratal) [reply Source/ARTPush.m:182]: This is in the context of `didRegisterForRemoteNotificationsWithDeviceToken` which is already APNS, so I think it's okay here.

**Review by lukasz-szyszkowski — APPROVED** [review]

**Review by ben-xD — APPROVED** [review]

**Comment by ben-xD** [comment]
Thanks, it's less confusing now :)

**Pull request merged**